electron/execution.ts

```diff
@@ -154,48 +167,59 @@ async function onTest(data, browserWindow) {
     });
   };
 
-  // returns TestEvent interface defined in common/types.ts
-  const constructEvent = parsed => {
+  const constructEvent = (parsed: Record<string, any>): TestEvent | null => {
     switch (parsed.type) {
```

Comment:
I think if we did something like this we can improve this function a bit. Reduces the number of ternary/return statements and gives greater confidence and static typing for each return. Also pushes the verification logic into a separate function, which will make it easier to write tests for.

```typescript
  const isJourneyStart = (event: any): event is { journey: { name: string } } =>
    event.type === 'journey/start';

  const isStepEnd = (
    event: any
  ): event is {
    step: { duration: { us: number }; name: string; status: StepStatus };
    error?: Error;
  } => {
    return (
      event.type === 'step/end' &&
      ['succeeded', 'failed', 'skipped'].includes(event.step?.status) &&
      typeof event.step?.duration?.us === 'number'
    );
  };

  const isJourneyEnd = (
    event: any
  ): event is { journey: { name: string; status: 'succeeded' | 'failed' } } => {
    return event.type === 'journey/end' && ['succeeded', 'failed'].includes(event.journey?.status);
  };

  const constructEvent = (parsed: Record<string, any>): TestEvent => {
    if (isJourneyStart(parsed)) {
      return {
        event: 'journey/start',
        data: {
          name: parsed.journey.name,
        },
      };
    } else if (isStepEnd(parsed)) {
      return {
        event: 'step/end',
        data: {
          name: parsed.step.name,
          status: parsed.step.status,
          duration: Math.ceil(parsed.step.duration.us / 1000),
        },
      };
    } else if (isJourneyEnd(parsed)) {
      return {
        event: 'journey/end',
        data: {
          name: parsed.journey.name,
          status: parsed.journey.status,
        },
      };
    }
    return null;
  };
```

Reply:
Thanks, Justin - this is great! After reading this code, I understand the reason you suggested using `unknown` for the parsed(event in here) :) I really like how you used the type predicate here. I will also change the `any` to `unknown` as it works nicely. 🙏

Reply:
Sounds fine - if we change `any` to `unknown` in the type guard functions, we might need to add some additional code to them to satisfy typescript as we are referencing fields on an `unknown` var now. I'm happy to pair on re-writing anything that doesn't drop right in.

Reply:
Thanks, I personally think it should be fine with `any` in this case